src/main/cpp/win.cpp

```diff
@@ -1133,4 +1133,111 @@ Java_net_rubygrapefruit_platform_internal_jni_WindowsRegistryFunctions_getValueN
     return true;
 }
 
+// File watching
+
+JavaVM* jvm = NULL;
+
+typedef struct watch_details {
+    HANDLE watchHandle;
+    HANDLE threadHandle;
+    jstring path;
+    jobject watcherCallback;
+} watch_details_t;
+
+DWORD WINAPI EventProcessingThread(LPVOID data) {
+    watch_details_t *details = (watch_details_t*) data;
+    while (TRUE) {
+        if (WaitForSingleObject(details->watchHandle, INFINITE) == WAIT_FAILED) {
+            // TODO Error handling
+            // mark_failed_with_errno(env, "could not wait for change notification", result);
+            break;
+        }
+        if (!FindNextChangeNotification(details->watchHandle)) {
+            if (GetLastError() == ERROR_INVALID_HANDLE) {
+                // Assumed closed
+                break;
+            }
+            // TODO Error handling
+            // mark_failed_with_errno(env, "could not schedule next change notification", result);
+            break;
+        }
+
+        JNIEnv* env;
+        int getEnvStat = jvm->GetEnv((void **)&env, JNI_VERSION_1_6);
```

Comment:
Extract to generic.h in order to shared between Mac/Windows.

Reply:
Moved it to `win.h`.